circuits/cpp/src/aztec3/utils/msgpack_derived_equals.hpp

```diff
@@ -0,0 +1,29 @@
+#pragma once
+#include <tuple>
+
+namespace aztec3::utils {
+// Auxiliary function for comparison of elements from tuples
+template <typename Tuple1, typename Tuple2, std::size_t... I>
+auto _compare_msgpack_tuples(const Tuple1& t1, const Tuple2& t2, std::index_sequence<I...>) // NOLINT
+{
+    // Compare every 2nd value from our key1, value1, key2, value2 list
+    return ((std::get<I * 2 + 1>(t1) == std::get<I * 2 + 1>(t2)) && ...);
```

Comment:
`&& ...`

C++'s dark magics never cease to amaze me

Reply:
left as an exercise to the compiler